interfaces/builtin/audit_control.go

```diff
@@ -0,0 +1,70 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const auditControlSummary = `allows control over the kernel audit system`
+
+const auditControlBaseDeclarationSlots = `
+  audit-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const auditControlConnectedPlugSecComp = `
+# Description: Can use netlink to communicate with kernel audit system.
+bind
+socket AF_NETLINK - NETLINK_AUDIT
+`
+
+const auditControlConnectedPlugAppArmor = `
+# Description: Can use netlink to communicate with kernel audit system.
+network netlink,
+
+# CAP_AUDIT_CONTROL required to enable/disable kernel auditing, change auditing
+# filter rules, and retrieve auditing status and filtering rules.
+capability audit_control,
+
+# Allow reading the login UID and session ID of processes.
+@{PROC}/*/{loginuid,sessionid} r,
+
+# Allow writing /run/auditd.pid and /run/auditd.state, as required by auditd.
```

Comment:
Since this interface seems to allow a snapped auditd - should it instead be named `auditd` or `auditd-support` instead (similar to `lxd-support` etc)?

Reply:
I agree `audit-control` isn't really correct in the usual way `*-control` interfaces work in snapd. The motivation for creating this interface is to enable a customer which wants to ship a snap which includes `auditd` and `rsyslog`, but the interface isn't designed to only support that snap, it should be something which is usable by any snap which needs the `audit_control` capability.

So perhaps access to `/run/auditd.pid` and `/run/auditd.state` should be removed here, and that particular customer should get write access to them via system files? I just don't know what applications on Linux use the `audit_control` capability other than `auditd/auditctl`.

Perhaps the same could be said for `@{PROC}/*/{loginuid,sessionid}` as well, we could remove this from the interface and have the customer get it via `system-files`?

Reply:
Discussed some with the team, we agreed to rename the interface to `auditd-support` and grant access to the filepaths related to auditd, rather than have the customer snap get them separately via `system-files`.